### PR TITLE
Add the block data api plugin

### PR DESCRIPTION
**Pull request by ingeniumed**
## Description

The purpose of this PR is to add the block data api plugin into the bundle. Since the block data api plugin has a rest and graphQL endpoint, this will now enable both of the APIs with this bundle.

The eventual goal would be to replace the bespoke the `WPGraphQL Content blocks` plugin, in favour of this plugin. The first step is add the plugin in, and then slowly add in any missing features so we can eventually deprecate it.

To review this PR, the changes that have been made:

- Added the plugin using `git subtree`:

`git remote add -f block-data-api-remote git@github.com:Automattic/vip-block-data-api.git`
`git subtree add --prefix lib/vip-block-data-api block-data-api-remote trunk --squash`

The following is how you update it:

`git fetch block-data-api-remote trunk`
`git subtree pull --prefix lib/vip-block-data-api block-data-api-remote trunk --squash`

- Added the block data api as a setting that can be turned on and off within the `settings.php`
- Required the plugin if its enabled within the `vip-decoupled.php`
- Updated the README with this new plugin with potential deprecation plans.

## Steps to Test

- Load up the dev-env or wp-env and enable this plugin
- Ensure the graphQL endpoint can be hit, using the query:

```gql
query AllPosts {
  posts {
    nodes {
      id
      title
      contentBlocks {
        blocks {
          attributes {
            name
            value
          }
          name
          innerHTML
        }
        isGutenberg
        version
      }
    }
  }
}
```

- Ensure the `wp-json/vip-block-data-api/v1/posts/1/blocks` rest endpoint can be hit

### CODE REVIEW

**Comment by alecgeatches** [comment]
Tested this locally, works great! Used this query to test the Block data WPGraphQL integration as well:

```graphql
query AllPosts {
  posts {
    nodes {
      id
      title
      blocksData {
        blocks {
          name
          attributes {
            name
            value
          }
        }
      }
    }
  }
}
```

After setting permalinks, the REST API at `wp-json/vip-block-data-api/v1/posts/1/blocks` worked properly too. Looks good to me!

File: vip-decoupled.php [hunk 1]
```diff
@@ -43,6 +43,13 @@
 	require_once __DIR__ . '/blocks/blocks.php';
 }
 
+/**
+ * VIP Block Data API.
+ */
+if ( is_plugin_enabled( 'block-data-api' ) ) {
```
Comment (chriszarate) [inline vip-decoupled.php:49]: Is there a chance the Block Data API also lands in `mu-plugins`? If so, and you want to prefer the mu-plugins version, you might want to proactively add some logic here to determine which version "wins."

File: README.md [hunk 2]
```diff
@@ -125,12 +125,46 @@ This will allow you to easily map Gutenberg blocks to front-end components. Post
...
 
+*Note: In a future update, this plugin will be deprecated and eventually removed in favour of the VIP Block Data API plugin.*
+
 ### WPGraphQL Preview
 
 This plugin overrides WordPress's native preview functionality and securely sends you to your decoupled frontend to preview your content. This ensures that your preview content has parity with your published content. It works by issuing a one-time use token, locked to a specific post, that can be redeemed by the frontend to obtain preview content for that post.
 
 **This plugin currently only works with our Next.js boilerplate** and should be disabled if you are not using it. If you are interested in using this plugin for other frontend frameworks, please see the [preview `README`][preview-readme].
 
+### VIP Block Data API
```
Comment (chriszarate) [inline README.md]: You might want to put both options next to each other and add a "Which should I choose?" section. That section can even just say, "we already recommend the block data API but here are some caveats"

Comment (ingeniumed) [reply README.md]: Great point, thanks for pointing that out. I've added this, and added what I think would be helpful in making this decision. Lemme know what you think.

**Comment by alecgeatches** [comment]
Note here - we just merged a change into the Block Data API and [released version `1.2.0`](https://github.com/Automattic/vip-block-data-api/releases/tag/1.2.0). I've merged those changes into this PR.

File: vip-decoupled.php (hunk 1 above)
Comment (alecgeatches) [reply vip-decoupled.php:49]: Good question! We have that logic built-in. In the plugin itself [we verify that we're not double-loading](https://github.com/Automattic/vip-decoupled-bundle/blob/7a8401c0146699e18ac2412bfdd268f5e0a2d520/lib/vip-block-data-api/vip-block-data-api.php#L20). On the `mu-plugins` integration side, we also have logic to [prefer the customer version of the plugin](https://github.com/Automattic/vip-go-mu-plugins/blob/d0af967/integrations/block-data-api.php#L39), if installed.

I spent some time just now testing what happens if the Block Data API plugin is installed, as well as the bundle with the Block Data API feature enabled, as well as the `mu-plugins` version. The priority goes:

1. Plugin version first
2. Bundle version second, if the plugin is disabled (second alphabetically)
3. The `mu-plugins` version of the plugin, if everything else is disabled

This fits our original intent to prefer customer code first, and works fine between the bundle and normal Block Data API plugin as well. Everything is working as designed.

**Review by alecgeatches — APPROVED** [review]
Approving for the merge. Thank you @ingeniumed for the integration and documentation!

**Pull request merged**